Not able to set deadline in grpc-web client call

I have a simple unary server; Protobuffer definition  https://github.com/alexcpn/go_grpc_2022/blob/main/interfaces/microservice_1/test.proto

```
service SearchService {
  // Method to Search a test service
  rpc Search (SearchRequest) returns (SearchResponse) {}
}
```

I am generating a grpc-web client using the following Make file as in 

```
compile_grpcweb:
	-mkdir -p generated
	protoc  -I ../../interfaces/ ../../interfaces/microservice_1/*.proto \
	--js_out=import_style=commonjs,binary:./generated \
	--grpc-web_out=import_style=typescript,mode=grpcweb:./generated
````
https://github.com/alexcpn/go_grpc_2022/blob/c8381e2ceb538cfce7be78089740419f2df1083a/microservice_1/ts_client/Makefile#L40-L44


The generated file has only two signatures

```
 search(
    request: microservice_1_test_pb.SearchRequest,
    metadata: grpcWeb.Metadata | null): Promise<microservice_1_test_pb.SearchResponse>;

  search(
    request: microservice_1_test_pb.SearchRequest,
    metadata: grpcWeb.Metadata | null,
    callback: (err: grpcWeb.RpcError,
               response: microservice_1_test_pb.SearchResponse) => void): grpcWeb.ClientReadableStream<microservice_1_test_pb.SearchResponse>;

```
https://github.com/alexcpn/go_grpc_2022/blob/main/microservice_1/ts_client/generated/microservice_1/TestServiceClientPb.ts

and in both not able to set deadline as mentioned in the documentation

```
    
    var dl = new Date();
    dl.setSeconds(dl.getSeconds() + 1);

   var result = this.searchService.search(searchRequest,{deadline:dl.getTime()})
    result.then(response => {
          this.responseData.data =response.getSearchResponse()
    }).catch(error  =>{
      console.log("Got error from server: grpcWeb.RpcError).message-->",(error as grpcWeb.RpcError).message)
    })

   const call =    this.searchService.search(searchRequest,{deadline:dl.getTime()},  (err: grpcWeb.RpcError, response: SearchResponse) =>  {
      if (err) {
        console.error("GRPC Error", err);
        this.responseData.error = err;
        this.searchService = <SearchServiceClient>{};
        return;
      }
      this.responseData.data =  response.getSearchResponse();
      console.log("Grpc-greeter.service.ts Response -->", this.responseData.data);
    });

```
The output of ng build

```
Error: src/app/grpc-greeter.service.ts:37:58 - error TS2322: Type 'number' is not assignable to type 'string'.

37    var result = this.searchService.search(searchRequest,{deadline:dl.getTime()})
                                                            ~~~~~~~~


Error: src/app/grpc-greeter.service.ts:44:61 - error TS2322: Type 'number' is not assignable to type 'string'.

44    const call =    this.searchService.search(searchRequest,{deadline:dl.getTime()},  (err: grpcWeb.RpcError, response: SearchResponse) =>  {
                                                               ~~~~~~~~

```

--- a/README.md
+++ b/README.md
@@ -225,7 +225,7 @@ should be a Unix timestamp, in milliseconds.
 var deadline = new Date();
 deadline.setSeconds(deadline.getSeconds() + 1);
 
-client.sayHelloAfterDelay(request, {deadline: deadline.getTime()},
+client.sayHelloAfterDelay(request, {deadline: deadline.getTime().toString()},
   (err, response) => {
     // err will be populated if the RPC exceeds the deadline
     ...

--- a/javascript/net/grpc/web/grpcwebclientbase_test.js
+++ b/javascript/net/grpc/web/grpcwebclientbase_test.js
@@ -84,8 +84,8 @@ testSuite({
     deadline.setSeconds(deadline.getSeconds() + 1);
     await new Promise((resolve, reject) => {
       client.rpcCall(
-          'url', new MockRequest(), {'deadline': deadline}, methodDescriptor,
-          (error, response) => {
+          'url', new MockRequest(), {'deadline': deadline.getTime().toString()},
+          methodDescriptor, (error, response) => {
             assertNull(error);
             resolve();
           });

--- a/test/interop/interop_client.js
+++ b/test/interop/interop_client.js
@@ -52,8 +52,10 @@ function doEmptyUnary(done) {
 
 function doEmptyUnaryWithDeadline(done) {
   var testService = new TestServiceClient(SERVER_HOST, null, null);
-  const deadlineMs = 1000; // 1 second
-  testService.emptyCall(new Empty(), {deadline: Date.now() + deadlineMs},
+
+  const deadline = new Date();
+  deadline.setSeconds(deadline.getSeconds() + 1);
+  testService.emptyCall(new Empty(), {deadline: deadline.getTime().toString()},
     (err, response) => {
       assert.ifError(err);
       assert(response instanceof Empty);

